src/Examine.AzureDirectory/AzureReadOnlyDirectory.cs

```diff
@@ -0,0 +1,322 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using Examine.LuceneEngine.Directories;
+using Lucene.Net.Index;
+using Lucene.Net.Store;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Directory = Lucene.Net.Store.Directory;
+
+namespace Examine.AzureDirectory
+{
+    public class AzureReadOnlyDirectory : ExamineDirectory
```

Comment:
Sorry, i guess i should have reviewed this one before reviewing the back porting of this. How come this needs to exist? It's a ton of duplicated code and then what is the point of having the isReadOnly flag in AzureDirectory?

Reply:
@Shazwazza I didnt removed flag from original directory, but that directory is strongly modified to use aliasing. It need exist to be explicit and avoid synchronization issues when you try read and write in same time, which doesnt really work when you have 2 sources of true. That is working in similar way as Elastic underhood which means using 2 different indexes to read and write, but we dont one storage, we have 2 storages there and we need sync them, it is why I am using aliasing in ReadOnlyDirectory.